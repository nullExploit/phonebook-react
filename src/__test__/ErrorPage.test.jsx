import { render } from "@testing-library/react";
import { createMemoryRouter, RouterProvider } from "react-router-dom";
import Providers from "../lib/redux/Provider";
import ErrorPage from "../components/ErrorPage";

describe("Should render error correctly", () => {
  it("Should render error", () => {
    const router = createMemoryRouter([{ path: "/", element: <ErrorPage /> }]);

    const { baseElement } = render(
      <Providers>
        <RouterProvider router={router} />
      </Providers>
    );

    expect(baseElement).toMatchSnapshot();
    expect(document.querySelector("h1").innerHTML).toBe("Oops!");
    expect(document.querySelector("p").innerHTML).toBe(
      "Sorry, an unexpected error has occurred."
    );
  });
});
