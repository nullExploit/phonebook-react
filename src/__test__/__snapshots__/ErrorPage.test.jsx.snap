// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render error correctly Should render error 1`] = `
<body>
  <div>
    <div
      class="container"
      id="error-page"
    >
      <h1>
        Oops!
      </h1>
      <p>
        Sorry, an unexpected error has occurred.
      </p>
      <p>
        <i />
      </p>
    </div>
  </div>
</body>
`;
